src/components/SearchModal/filtering.ts

```diff
@@ -42,3 +43,39 @@ export function filterTokens(tokens: Token[], search: string): Token[] {
   //   return 0
   // })
 }
+
+export function useSortedTokensByQuery(tokens: Token[] | undefined, searchQuery: string): Token[] {
+  return useMemo(() => {
+    if (!tokens) {
+      return []
+    }
+
+    const symbolMatch = searchQuery
+      .toLowerCase()
+      .split(/\s+/)
+      .filter(s => s.length > 0)
+
+    if (symbolMatch.length > 1) {
+      return tokens
+    }
+
+    return [
+      // sort any exact symbol matches first
+      ...tokens.filter(token => token.symbol?.toLowerCase() === symbolMatch[0]),
+
+      // sort by tokens whos symbols start with search substrng
+      ...tokens.filter(
+        token =>
+          token.symbol?.toLowerCase().startsWith(searchQuery.toLowerCase().trim()) &&
+          token.symbol?.toLowerCase() !== symbolMatch[0]
+      ),
+
+      // rest that dont match upove
+      ...tokens.filter(
+        token =>
+          !token.symbol?.toLowerCase().startsWith(searchQuery.toLowerCase().trim()) &&
+          token.symbol?.toLowerCase() !== symbolMatch[0]
+      )
+    ]
```

Comment:
minor nit - perf could be improved here by calling filter on the subset of tokens not chosen by prior calls, instead of re-filtering the whole list, and by only doing the `toLowerCase` and `trim` on first iteration (and once at the start for the `searchQuery`.

Reply:
Will take a look here - the filtering is kinda a mess, theres 3 levels to it right now